x/emissions/keeper/actor_utils/worker.go

```diff
@@ -141,19 +140,14 @@ func insertInferencesFromTopInferers(
 	inferencesToInsert := types.Inferences{
 		Inferences: inferences,
 	}
-	err := k.InsertInferences(ctx, topicId, nonce, inferencesToInsert)
+	err := k.InsertInferences(ctx, topicId, nonce.BlockHeight, inferencesToInsert)
 	if err != nil {
 		return nil, err
 	}
 
 	return acceptedInferers, nil
 }
 
-// Output a new set of forecasts where only 1 forecast per registered forecaster is kept,
-// ignore the rest. In particular, take the first forecast from each registered forecaster
-// and none from any unregistered forecaster.
-// Signatures, anti-synil procedures, and "skimming of only the top few workers by score
-// descending" should be done here.
```

Comment:
we should have some function description here if only for the principle of the thing